modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/AddEndPointSecurityPerTypeTestCase.java

```diff
@@ -0,0 +1,476 @@
+
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN},
+                new Object[]{TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void initialize() throws Exception {
+
+        super.init(userMode);
+        dcrURL = backEndServerUrl.getWebAppURLHttp()+ "client-registration/v0.17/register";
+        apiEndPointUrl = getGatewayURLNhttp() + "backendSecurity/1.0.0";
+        providerName = user.getUserName();
+        requestHeadersGet = new HashMap<>();
+        requestHeadersGet.put("accept", "text/plain");
+        requestHeadersGet.put("Content-Type", "text/plain");
+
+        //Create application
+        ApplicationDTO dto = restAPIStore.addApplicationWithTokenType(APPLICATION_NAME,
+                APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, "", "",
+                ApplicationDTO.TokenTypeEnum.OAUTH.getValue());
+        applicationID = dto.getApplicationId();
+        ArrayList<String> grantTypes = new ArrayList<>();
+        grantTypes.add(APIMIntegrationConstants.GRANT_TYPE.CLIENT_CREDENTIAL);
+        productionApplication = restAPIStore
+                .generateKeys(applicationID, "3600", null, ApplicationKeyGenerateRequestDTO.KeyTypeEnum.PRODUCTION,
+                        null, grantTypes);
+        sandboxApplication = restAPIStore
+                .generateKeys(applicationID, "3600", null, ApplicationKeyGenerateRequestDTO.KeyTypeEnum.SANDBOX,
+                        null, grantTypes);
+        DCRParamRequest oauthAppProduction = new DCRParamRequest("dummyapp--production", null, null,
+                user.getUserName(), "password client_credentials", dcrURL, user.getUserName(),
+                user.getPassword(), null);
+        DCRParamRequest oauthAppSandbox = new DCRParamRequest("dummyapp--sandbox", null, null, user.getUserName(),
+                "password client_credentials", dcrURL, user.getUserName(), user.getPassword(), null);
+        applicationKeyBeanProduction = ClientAuthenticator.makeDCRRequest(oauthAppProduction);
+        applicationKeyBeanSandbox = ClientAuthenticator.makeDCRRequest(oauthAppSandbox);
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Add Endpoint Security for production")
+    public void testAddEndpointSecurityForProduction() throws Exception {
+
+        // Create an API
+        String apiName = API_NAME + "1";
+        String apiContext = API_CONTEXT + "1";
+        APICreationRequestBean apiCreationRequestBean =
+                new APICreationRequestBean(apiName, apiContext, API_VERSION_1_0_0, providerName,
+                        new URL(apiEndPointUrl), new URL(apiEndPointUrl));
+        apiCreationRequestBean.setTier(TIER_UNLIMITED);
+        apiCreationRequestBean.setTiersCollection(TIER_UNLIMITED);
+        apiCreationRequestBean.setProvider(user.getUserName());
+        apiCreationRequestBean.setSetEndpointSecurityDirectlyToEndpoint(Boolean.TRUE);
```

Comment:
Is setting this FALSE (or not setting it) already handled in other test cases?

Reply:
Yes, the only place this function will be used is inside this test case file by setting this as TRUE. 
By default this value **setSetEndpointSecurityDirectlyToEndpoint** will have FALSE [1]. That means, in every other place this will be FALSE so that the else part of the if statement in [2] will get executed which aligns with the original behaviour we had before adding this new parameter.

[1] https://github.com/wso2/product-apim/pull/11444/files#diff-987c398ba7ceb48bac9e57af64a5953d3dad4dc95edbb5bdc44ac2c3a43b8c2fR110
[2] https://github.com/wso2/product-apim/pull/11444/files#diff-e2b0209eebfea247bd6a651abdc6ce6f47f32fa2e00f311fa7a34d76cd0404c5R1186